### ISSUE TITLE
Add "depends_on" field to the resources (modules)

### ISSUE THREAD

**Issue by ashkolte** [GoogleCloudPlatform/healthcare-data-protection-suite#753]
Resources like Cloud_sql, GKE, Bigquery, GCS Bucket etc does not support "depends_on" field (Natively supported in terraform). This causes resources creation failure due to dependencies.

Example: While Bigquery API is getting enabled, Bigquery dataset creation failed with API disabled error.

**Comment by umairidris** [comment]
Hi @ashkolte,

I believe https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/pull/756 will indirectly resolve the issues faced by this issue. Please confirm once it is merged into the repo.

I will demote priority on this and likely not get to it next week. If this continues to block your usage please let me know and I can circle back on it. Thanks!

**Comment by xingao267** [comment]
@ashkolte is the issue fixed?

**Comment by ashkolte** [comment]
Yes

**Comment by xingao267** [comment]
Thanks for confirming @ashkolte
